lemkus_spider.py

```diff
@@ -0,0 +1,59 @@
+import json
+import re
+import scrapy
+
+
+class LemkusSpider(scrapy.Spider):
+    name = "lemkus-spider"
+    start_urls = [
+        'https://lemkus.com/collections/sneakers',
+        'https://lemkus.com/collections/apparel',
+        'https://lemkus.com/collections/kids'
+    ]
+
+    @staticmethod
+    def get_current_page_no_to_scrap(response):
+        current_page_url = response.css('link[rel="canonical"]::attr(href)').extract()[0].split('?')
+        if len(current_page_url) < 2:
+            return 1
+        return int(current_page_url[-1].split("=")[-1])
+
+    @staticmethod
+    def get_next_page_no_to_scrap(response):
+        next_page_url = response.css('link[rel="next"]::attr(href)').extract()
+        if len(next_page_url) > 0:
+            return int(next_page_url[0].split('?')[-1].split("=")[-1])
+        return None
+
+    @staticmethod
+    def get_product_collection_id(response):
+        collection_id_pattern_in_response = re.compile('var page_id = (.*?);')

```

Comment:
Compiling a regex at every function call is not going to give any specific benefit. Declare a class level variable with compiled regex.